subsys/net/ip/tcp2.c

```diff
@@ -37,17 +37,18 @@ size_t (*tcp_recv_cb)(struct tcp *conn, struct net_pkt *pkt) = NULL;
 
 static struct tcphdr *th_get(struct net_pkt *pkt)
 {
-	NET_PKT_DATA_ACCESS_DEFINE(th_access, struct tcphdr);
```

Comment:
Hmm, not sure about this. If the header is split between two net_buf's, then this will not work as expected.
I think better would be to move the th_access to calling function, and then pass pointer to it here and use it here as is.

Reply:
We do a check if the TCP header is contiguous with net_pkt_is_contiguous().

Do you think that the case that the TCP header is split is realistic?

Reply:
Well, difficult to say how realistic it is, probably not very. But anyway, the original code would work with split header but after the changes, the func would return NULL if that happens.

Reply:
Anyway, my point was that it is quite simple to make this work properly by just moving the th_access to calling function.

Reply:
Hmm, we call th_get() from multiple functions.

Reply:
Yes, that is unfortunate and causes many changes. We have spent lot of time to make sure that the networking code works if for some reason the IP/TCP/UDP header is split between two network packet. I do not wish that we start to slip from this goal even if it very unlikely that the header would be split between two net_bufs.

Reply:
This works but only if `net_buf->size >= sizeof(tcp_hdr)`. This is certainly the case atm so I suppose we can continue with this solution.